File: landing/public/examples/js/project/physics.js
// Physics helpers shared with main.js through the playground's single
// global scope (this file is listed before the entry).
"use strict";

function stepBall(b, w, h) {
  b.x += b.vx;
  b.y += b.vy;
  if (b.x < 8 || b.x > w - 8) b.vx = -b.vx;
  if (b.y < 8 || b.y > h - 8) b.vy = -b.vy;
}

function inside(px, py, x, y, w, h) {
  return x <= px && px < x + w && y <= py && py < y + h;
}

File: examples/python/project/physics.py
# Integer physics helpers imported by main.py.

def step_ball(b, w, h):
    b[0] = b[0] + b[2]
    b[1] = b[1] + b[3]
    if b[0] < 8 or b[0] > w - 8:
        b[2] = -b[2]
    if b[1] < 8 or b[1] > h - 8:
        b[3] = -b[3]

def inside(px, py, x, y, w, h):
    return x <= px and px < x + w and y <= py and py < y + h
